src/utils/fetchEthValidatorData.ts

```diff
@@ -0,0 +1,35 @@
+import { ethers } from 'ethers';
+
+interface EthValidatorPerformanceData {
+  status: string;
+  data: {
+    balance: number;
+    performance1d: number;
+    performance31d: number;
+    performance365d: number;
+    performance7d: number;
+    performancetoday: number;
+    performancetotal: number;
+    rank7d: number;
+    validatorindex: number;
+  }[];
+}
+const ETH_VALIDATOR_PERFORMANCE_URL = 'https://beaconcha.in/api/v1/validator/402418/performance';
+
+export const fetchEthValidatorTotalPerformance = async () => {
+  try {
+    const data = await fetch(ETH_VALIDATOR_PERFORMANCE_URL);
+    const ethValidatorData: EthValidatorPerformanceData = await data.json();
+    if (ethValidatorData.status !== 'OK') {
+      return null;
+    }
+    return {
+      totalPerformanceEther: ethers.utils.formatUnits(
+        ethValidatorData.data[0].performancetotal.toString(),
+        'gwei'
+      ),
+    };
+  } catch (e) {
+    console.error('> fetchEthValidatorTotalPerformance', e);
+  }
```

Comment:
If we return null or undefined when an error arrises, we'd be setting the local stored value to those values since this is the code calling this function:

```    
validatorPerformance = {
      eth: await getEthValidatorPerformance(),
      ftm: await getFtmValidatorPerformance(),
      fuse: await getFuseValidatorPerformance(),
    };
```

Reply:
I'd suggest we throw an error when something happens and then work with the PromiseSettledResult from all the 3 performance calls, updating the stored value only for those calls that succeeded